### PR TITLE
feature(cdcmultidc): New job 8h-cdc-multidc-topology-changes

### PR DESCRIPTION
**Pull request by aleksbykov**
Starting from branch-4.6 new system_distributed_everywhere keyspace was added. 
Data of this keyspace replicated to each node in cluster. First consumer of the keyspace
is cdc. It stores there new cdc_generation_description table. It is system table and not 
should be used by user. Regular tests for cdc + multi dc should be enough for testing.

Add new job for testing cdc in multi dc cluster configuration with new nemesis
Update c-s profiles for cdc to support multi dc.
Change replication class to NetworkTopology

Trello: https://trello.com/c/3xwu3ugW

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/nemesis.py
Comment (roydahan) [inline sdcm/nemesis.py:4158]: Why do we still need this and can't use "SisyphusMonkey" with nemesis_include_filter: ['topology_changes']?

File: test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml
Comment (roydahan) [inline test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml:15]: This should be SisyphusMonkey as I wrote above.

File: test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml
Comment (roydahan) [inline test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml:24]: Why do we need this? CDC isn't experimental.

File: test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml
Comment (roydahan) [inline test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml:23]: ?

File: test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml
Comment (roydahan) [inline test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml:10]: maybe 2xlarge is enough?
Is the load high?

**Review by roydahan — CHANGES_REQUESTED** [review]

File: sdcm/nemesis.py
Comment (dkropachev) [inline sdcm/nemesis.py:366]: @aleksbykov , let's get rid of flag_list and do just following:
```suggestion
        subclasses_list = self._get_subclasses(
            disruptive=disruptive,
            run_with_gemini=run_with_gemini,
            networking=networking,
            kubernetes=kubernetes,
            limited=limited,
            topology_changes=topology_changes
        )
```

**Review by dkropachev — CHANGES_REQUESTED** [review]

**Comment by soyacz** [comment]
This test relates to warning in logs like:
```
Jan 25 10:26:45 r1.tab.com scylla[31756]:  [shard 0] seastar_memory - oversized allocation: 2023424 bytes. This is non-fatal, but could lead to latency and/or fragmentation issues. Please report: ...
```
 (from https://github.com/scylladb/scylla/issues/7961)

Shouldn't we catch `oversized allocation` warnings in SCT tests?

**Comment by roydahan** [comment]
> This test relates to warning in logs like:
> 
> ```
> Jan 25 10:26:45 r1.tab.com scylla[31756]:  [shard 0] seastar_memory - oversized allocation: 2023424 bytes. This is non-fatal, but could lead to latency and/or fragmentation issues. Please report: ...
> ```
> 
> (from [scylladb/scylla#7961](https://github.com/scylladb/scylla/issues/7961))
> 
> Shouldn't we catch `oversized allocation` warnings in SCT tests?

Yes, we should.
Even as ERROR and not warning, but we should have a way to differ between oversized allocations that we know of and such like above that were fixed.

File: sdcm/nemesis.py
Comment (aleksbykov) [reply sdcm/nemesis.py:4158]: fixed

File: test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml
Comment (aleksbykov) [reply test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml:15]: fixed

File: test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml
Comment (aleksbykov) [reply test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml:24]: fixed

File: test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml
Comment (aleksbykov) [reply test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml:23]: remove experimental

File: test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml
Comment (aleksbykov) [reply test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml:10]: load is high, and also we have at 4 table with cdc enabled. Smaller instances could be not enough

File: sdcm/nemesis.py
Comment (roydahan) [inline sdcm/nemesis.py:3802]: I'm not sure it's considered as a topology change, but I'm ok to leave it to be on the safe side.

**Review by roydahan — DISMISSED** [review]

**Comment by roydahan** [comment]
@aleksbykov check the unittests that are failing and get another maintainer to review.

File: sdcm/nemesis.py
Comment (dkropachev) [inline sdcm/nemesis.py:144]: Please remove left over

File: sdcm/nemesis.py
Comment (dkropachev) [inline sdcm/nemesis.py:142]: Let's start adding description to the flag, so that you could understand what it represents.
Also please accumulate flags in one place and add : bool to them

**Review by dkropachev — CHANGES_REQUESTED** [review]
Also please separate commit regarding `topology_changes` and  test files

File: sdcm/nemesis.py
Comment (aleksbykov) [reply sdcm/nemesis.py:144]: done

File: sdcm/nemesis.py
Comment (aleksbykov) [reply sdcm/nemesis.py:142]: done

File: sdcm/nemesis.py
Comment (dkropachev) [inline sdcm/nemesis.py:137]:
```suggestion
    # nemesis flags:
```

File: sdcm/nemesis.py
Comment (dkropachev) [inline sdcm/nemesis.py:138]:
```suggestion
    topology_changes: bool = False  # flag that signal that nemesis is changing cluster topology, i.e. adding/removing nodes/data centers
```

**Review by dkropachev — CHANGES_REQUESTED** [review]

**Review by dkropachev — APPROVED** [review]
Unit test failed due to the flaky test, not related to this PR

File: test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml
Comment (roydahan) [inline test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml:25]: Why do you need this one?
Did you change something from default?

**Review by roydahan — COMMENTED** [review]
Small comment regarding the append_scylla_args which I don't think we should override in the yaml, unless there is a special change (then add a comment why).

File: test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml
Comment (aleksbykov) [reply test-cases/longevity/longevity-cdc-100gb-8h-multi-dc-topology-changes.yaml:25]: No. it should be default

**Comment by aleksbykov** [comment]
> Small comment regarding the append_scylla_args which I don't think we should override in the yaml, unless there is a special change (then add a comment why).

I agree with you. if i don't mistaken, we cant append scylla-args in yaml, we should provide full.

**Review by roydahan — APPROVED** [review]

**Pull request merged**
